docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +164,104 @@ Note that:
+
+To manage configuration files for child-devices the `c8y_configuration_plugin` supports two aspects:
+1) Allowing to associate a configuration file with a cloud's child-device twin
+2) Allowing to consume/provide a configuration file from/to an external device via network
+
+**Details to Aspect 1: Associating to cloud's child-device twin**
+
+For aspect (1) the plugin provides the field `childid` for all records in the `c8y_configuration_plugin` configuration (reference to section 'Configuration' above). That field is interpreted as unique child-device id and the plugin associates the record's configuration file with corresponding cloud's child-device twin. If not provided the configuration file is associated with cloud's thin-edge device twin.
+
+Example:
+```shell
+$ cat /etc/tedge/c8y/c8y-configuration-plugin.toml
+files = [
+    { path = '/etc/tedge/tedge.toml', type = 'tedge.toml' },              # appears in the cloud on the thin-edge device
+    { path = '/etc/child1/foo.conf', type = 'foo', childid = 'child1'  }  # appears in the cloud on the child-device 'child1'
+  ]
+```
+
+Note that the `c8y_configuration_plugin` does not create any child-device twin in the cloud. Instead the clouds child-device twin must be created upfront. However the plugin takes care to set all capabilities (supported operations, config-types) on that child-device twin in the cloud in the same way as it does for the thin-edge device.
+
+
+**Details to Aspect 2: Consuming/providing configuration files via network**
+
+For aspect (2) there are two proposals as below. Decision has to been taken which proposal to follow.
+
+**Proposal 1:**
+
+The `c8y_configuration_plugin` configuration's record field `path` (reference to section 'Configuration' above) can be prefixed with "mqtt://". Then it's value is treated as MQTT topic structure, where another process/external device is expected to provide/consume the configuration file.
+
+The `c8y_configuration_plugin` expects the process/external device always putting the latest config file as retain message to the MQTT topic `{path}`. Then the plugin consumes that retain message whenever a cloud request comes in.
+
+Example Plugin Config:
+```shell
+$ cat /etc/tedge/c8y/c8y-configuration-plugin.toml
+files = [
+    { path = '/etc/tedge/tedge.toml', type = 'tedge.toml' },
+    { path = 'mqtt://configs/bar.conf', type = 'bar.conf', childid = 'child1'  }
+  ]
+```
+
+Example Flow:
+
+**Start Behaviour:**
+  * external device `child1`: starts
+  * external device `child1`: publishs all its config files to MQTT (with retain); Example: Publish to topic `configs/bar.conf`
+
+**Device-to-Cloud Behaviour:**
+  * at some point a config retrieval for type `bar.conf` for `child1` arrives at C8Y config plugin
+  * C8Y config plugin: subscribes to `configs/bar.conf` and gets immediately the config file content from the retained message
+  * C8Y config plugin: sends the recived MQTT payload as config file to C8Y<br/><br/>
```

Comment:
For parent device, the plugin uploads a file via HTTP. How the topic and MQTT payload to c8y will look like?

Reply:
Sorry, that sentence is missleading. I was trying to say, that the config file was received on MQTT, and is after it is sent to C8Y. So MQTT was meant for the _receiving_ part. Lets simplify that sentence.

```suggestion
  * C8Y config plugin: sends the config file to C8Y<br/><br/>
```

Reply:
> Just to clarify my understanding, the request could be for `upload/download` comes via MQTT but the actual `upload/download` of the file happens via HTTP right?.

We are talking here about two things: 

1) The file-transfer between the thin-edge device and C8Y
2) The file-transfer between the thin-edge device and an external-device (that represents the child-device)

Part 1 will be not changed for child-device support. It is currently HTTP and we are not about to change that.

Part 2 is currently not possible, cause the C8Y Config Plugin expects all config-files in the local filesystem. Here we need a way to transfer files from/to the external device. That section focuses on that file-transfer channel.

Please note, that section is one proposal for the file-transfer channel. There is also another proposal in that document (see "Proposal 2") , and also another proposal in the comments (see https://github.com/thin-edge/thin-edge.io/pull/1236#discussion_r926708254)